src/storage/mvcc/txn.rs

```diff
@@ -223,7 +236,23 @@ impl<S: Snapshot> MvccTxn<S> {
             }
         }
 
+        if let Some(secondary_keys) = secondary_keys {
+            lock.use_async_commit = true;
+            lock.secondaries = secondary_keys.to_owned();
+            // We will reuse min_commit_ts for async commit for now.
+            assert!(
+                min_commit_ts.is_zero(),
+                "async commit is not yet compatible with large transactions"
+            );
+
+            // TODO(nrc) this is going to block all the other keys' processing and writing, we should
```

Comment:
Will it be ok to set the `min_commit_ts` int the `PrewriteResponse` after the prewrite is done and schedule it as a future task into some `tso threads`?

Reply:
Yes, I would like to do something like that.